x-pack/solutions/observability/plugins/apm/public/hooks/use_environments_fetcher.tsx

```diff
@@ -51,6 +51,24 @@ export function useEnvironmentsFetcher({
       }).then((response) => {
         return { environments: response.terms };
       });
+
+      const unsetEnvironments = callApmApi('GET /internal/apm/environments/unset', {
+        params: {
+          query: {
+            start,
+            end,
+          },
+        },
+      }).then((response) => {
+        return response.hasUnsetEnvironment
+          ? { environments: [ENVIRONMENT_NOT_DEFINED_VALUE] }
+          : { environments: [] };
+      });
+      return Promise.all([suggestions, unsetEnvironments]).then((results) => {
+        return {
+          environments: results.map((result) => result.environments).flat(),
+        };
+      });
```

Comment:
Perhaps we could do is to run this whole thing in the `GET /internal/apm/suggestions` endpoint, eg:


```ts
const [hasUnset, envs] = Promise.all([hasUnsetEnvironments(), 
  serviceName ? getSuggestionsWithTermsEnum(...) : getSuggestionsWithTermsAggregation(...)
])

// merge hasUnset and envs and return it
```